src/Streamly/Internal/Data/Parser/ParserD.hs

```diff
@@ -439,13 +439,36 @@ sliceSepBy _cond = undefined
 
 -- | See 'Streamly.Internal.Data.Parser.sliceBeginWith'.
 --
--- /Unimplemented/
+-- /Internal/
 --
+data SliceBeginWithState s = Left' s | Right' s
+
 {-# INLINABLE sliceBeginWith #-}
-sliceBeginWith ::
-    -- Monad m =>
-    (a -> Bool) -> Fold m a b -> Parser m a b
-sliceBeginWith = undefined
+sliceBeginWith :: Monad m => (a -> Bool) -> Fold m a b -> Parser m a b
```

Comment:
Can you change the `INLINEABLE` to `INLINE` and run the benchmarks again?

Reply:
yep, it worked.